pkg/email/send_email.go

```diff
@@ -14,43 +14,29 @@ type Emailer struct {
 	Limiter *rate.Limiter
 }
 
-var (
-	ErrRateLimitExceeded = errors.New("rate limit exceeded")
-)
-
-const (
-	defaultRateLimit = 1.0 / 60.0
-	defaultBurstRate = 10
-)
+var ErrRateLimitExceeded = errors.New("rate limit exceeded")
 
 type EmailParams struct {
-	SMTPHost   string
-	Port       int
-	Username   string
-	Password   string
-	Sender     string
-	LimitEvery float64
-	Burst      int
+	SMTPHost           string
+	Port               int
+	Username           string
+	Password           string
+	Sender             string
+	LimitEveryDuration time.Duration
+	Burst              int
 }
 
 func NewEmailer(e EmailParams) Emailer {
```

Comment:
Prefer to return a pointer in this case. Interface that holds implementation or case where copy the Emailer value will not support value copy.